lib/pos_blockchain.go

```diff
@@ -134,8 +135,48 @@ func (bc *Blockchain) validateDeSoBlockPoS(desoBlock *MsgDeSoBlock) error {
 // that the timestamp is valid, that the version of the header is valid,
 // and other general integrity checks (such as not malformed).
 func (bc *Blockchain) validateBlockGeneral(desoBlock *MsgDeSoBlock) error {
-	// TODO: Implement me
-	return errors.New("IMPLEMENT ME")
+	// First make sure we have a non-nil header
+	if desoBlock.Header == nil {
+		return RuleErrorNilBlockHeader
+	}
+
+	// Timestamp validation
+	// What's the criteria? Is it just that it's not in the future?
+	// And that it's not too far in the past? Too far being older than the current best chain tip?
+	if desoBlock.Header.TstampNanoSecs < bc.GetBestChainTip().Header.TstampNanoSecs {
```

Comment:
A block can't have a timestamp that precedes its parent's timestamp. The parent may or may not be the current chain tip, so you need to explicitly look up the parent here.

If the parent doesn't exist, then the validation should fail here.
```suggestion
	if desoBlock.Header.TstampNanoSecs < bc.getParentBlock[desoBlock.Header.GetHash())] {
```